src/db/car/many.rs

```diff
@@ -9,29 +9,40 @@
 //! A single z-frame cache is shared between all read-only stores.
 
 use super::{AnyCar, ZstdFrameCache};
-use crate::blocks::Tipset;
 use crate::db::MemoryDB;
+use crate::libp2p_bitswap::BitswapStoreReadWrite;
 use crate::utils::io::random_access::RandomAccessFile;
+use crate::{blocks::Tipset, libp2p_bitswap::BitswapStoreRead};
 use anyhow::Context;
 use cid::Cid;
 use fvm_ipld_blockstore::Blockstore;
-use parking_lot::Mutex;
+use parking_lot::{Mutex, RwLock, RwLockReadGuard};
 use std::{io, path::PathBuf, sync::Arc};
 
 pub struct ManyCar<WriterT = MemoryDB> {
     shared_cache: Arc<Mutex<ZstdFrameCache>>,
     read_only: Vec<AnyCar<Box<dyn super::RandomAccessFileReader>>>,
-    writer: WriterT,
+    writer: RwLock<WriterT>,
```

Comment:
No need for this to be locked anymore.

```suggestion
    writer: WriterT,
```